src/main/java/org/jboss/jandex/TypeVariable.java

```diff
@@ -131,19 +142,19 @@ TypeVariable copyType(int boundIndex, Type bound) {
 
         Type[] bounds = this.bounds.clone();
         bounds[boundIndex] = bound;
-        return new TypeVariable(name, bounds, annotationArray());
+        return new TypeVariable(name, bounds, annotationArray(), hasImplicitObjectBound());
     }
 
     @Override
     public int hashCode() {
-        int hash = this.hash;
+        int hash = this.hash & Integer.MAX_VALUE;
         if (hash != 0) {
```

Comment:
Frankly speaking, this `MAX_VALUE` vs `MIN_VALUE` looks very hard to review.

What do you think:

a) If a regular `boolean` field was added? It would be way easier to understand.
b) Another option is to introduce a named constant for the bit. Currently, MIN vs MAX says nothing with regards to `implicitObjectBound`

Reply:
Ideally we would be able to use a boolean but unfortunately it costs us another 8 bytes per type variable, and these can easily be in 10-100k range of instances in large projects. Openjdk on x86-64 aligns objects on 8 byte boundary and sometimes you can get lucky if there is unused padding. However I checked and adding a boolean moves us from 32 bytes to 40 bytes per TypeVar (shallow space). Good point on b) I introduced intermediary constants